scipy/sparse/csgraph/_laplacian.py

```diff
@@ -34,8 +34,9 @@ def laplacian(csgraph, normed=False, return_diag=False, use_out_degree=False):
 
     Returns
     -------
-    lap : ndarray
-        The N x N laplacian matrix of graph.
+    lap : ndarray or coo_matrix
+        The N x N laplacian matrix of csgraph. It will be a numpy array (dense)
+        if the input was dense, sparse (COO format) otherwise.
```

Comment:
The problem with specifying sparse formats is exactly that: it enables users to make their own optimizations, which will then result in broken code if we choose to change the output format later.

It's much easier from a maintenance perspective to keep the docstrings vague and encourage users to have their code work with any sparse format, even if that means they just manually convert the result to a specific format.